lifx-core/src/lib.rs

```diff
@@ -1840,9 +1858,14 @@ impl FrameAddress {
     fn packed_size() -> usize {
         16
     }
-    fn validate(&self) {
-        //assert_eq!(self.reserved, [0;6]);
-        //assert_eq!(self.reserved2, 0);
+    fn validate(&self) -> Result<(), Error> {
+        if self.reserved != [0; 6] && self.reserved2 != 0 {
+            return Err(Error::RawMessage(
+                "reserved fields must be zeroed".to_string(),
+            ));
+        }
```

Comment:
This looks right in theory, but I don't think we can do this because I think these fields are sometimes non-zero.  When constructing packets, we (as 3rd-party developers) must set them to zero, but proprietary LIFX apps can have non-zero data here.  At least I think this is why I had the asserts commented out.

Reply:
Not sure how you want to handle this other than not returning an error but rather logging some sort of info message that indicates the unpacked reserved fields are not zeroed.

Reply:
What if the validate method took a `strict` argument or something?  If strict, the validation would check the reserved fields.  If not, skip those fields

```rust
fn validate(&self, strict: bool) -> Result<(), Error> {
  if strict {
    // check reserved fields
  }
  // check other fields
}